Velocity_Model/scripts/combine_basin_stats_z.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+
+out_file = "basin_stats_z.csv"
+
+
+from pathlib import Path
+import pandas as pd
+import argparse
+
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("basin_stats_csv", help="file containing (lon,lat,NZGD_lon,NZGD_lat,in_basin,basin_name) info. produced by gen_basin_stats.py")
+    parser.add_argument("basin_stats_z", help="file containing (name,Z1.0,Z2.5,sigma) info. produced by get_z.py")
+
+    args = parser.parse_args()
+
+
+    csv_df=pd.read_csv(args.basin_stats_csv,header=None,names=['lon','lat','NZGD_lon','NZGD_lat','in_basin','basin_name'])
+    z_df=pd.read_csv(args.basin_stats_z,names=['name','Z1.0','Z2.5','sigma'],index_col=0)
+
+
+    assert len(csv_df)==len(z_df)
+
+    #make a new column "name" buy combining two existing columns
+    csv_df["name"]=(csv_df["NZGD_lon"].map(str)+"_"+csv_df["NZGD_lat"].map(str))
+    csv_df=csv_df.set_index("name")
+
+
+    # We assume both df have the same set of names
+    csv_df["Z1.0"]=z_df["Z1.0"]
+    csv_df["Z2.5"]=z_df["Z2.5"]
```

Comment:
Is there an advantage/disadvantage to using a merge here, rather than just column assignment?

Reply:
@jpa198  Good advice. Just pushed the change.  The merge will make sure everything will work even if the rows in both DFs are not in the same order.

Reply:
As an FYI for large amount of rows this operation would be slow as pandas won't have an index (or categorical) over the name column.

Reply:
Well spotted. Already noticed the performance issue before you commented :)

https://github.com/ucgmsim/Velocity-Model/pull/44/commits/dba721b56fa7a9a4e9011056c4e75be63afff346

Reply:
I meant the performance of the merge, rather than the direct column overwrite as you originally were doing.

Reply:
I am not worried about the performance here.  The merge is almost instant. These two DFs are always in the exactly same order.